src/main/resources/assets/js/app/installation/view/MarketAppsTreeGrid.ts

```diff
@@ -199,15 +197,16 @@ export class MarketAppsTreeGrid
         this.updateApp(app);
     }
 
-    updateAppUninstalled(app: Application): void {
+    updateAppUninstalled(uninstalled: Application): void {
         this.installedApplications =
-            this.installedApplications.filter((app: Application) => app.getApplicationKey() !== app.getApplicationKey());
-        this.updateApp(app);
+            this.installedApplications.filter((app: Application) => app.getApplicationKey() !== uninstalled.getApplicationKey());
+        this.updateApp(uninstalled);
     }
 
     private updateApp(app: Application): void {
         this.updateAndSortApps(this.marketApps);
-        const marketApp = this.marketApps.find((marketApp: MarketApplication) => marketApp.getAppKey().equals(app.getApplicationKey()));
+        const marketApp = this.marketApps.find((marketApp: MarketApplication) => marketApp.getAppKey().equals(app.getApplicationKey()) ||
+            this.replaceDashesWithDotsInMarketApp(marketApp) === app.getApplicationKey().toString());
```

Comment:
We are doing exactly the same here as in `getInstalledAppByMarketApp`. We should have a separate function (`appsHaveMatchingKeys(app1, app2)`) where we compare the keys of two apps (with and without dashes).

Reply:
Those Application and MarketApplication switch roles in mentioned functions, thus the code won't be identical to them